### PR TITLE
Allow logging format configuration

### PR DESCRIPTION
**Pull request by ssbarnea**
This allows user to control log formatting strings.

Change-Id: I4c6aaddd1be241ec161be24571cf254d9217584d
Signed-off-by: Sorin Sbarnea <ssbarnea@redhat.com>

##### SUMMARY

This allows us to change the format of the log files which until now was hardcoded in ansible.

Relates to #13702 

##### ISSUE TYPE
 - Feature Pull Request

### CODE REVIEW

File: lib/ansible/config/base.yml [hunk 1]
```diff
@@ -669,6 +669,20 @@ DEFAULT_LOG_PATH:
...
+  ini:
+  - {key: log_format, section: defaults}
+  yaml: {key: defaults.log_format}
+DEFAULT_LOG_NAME:
+  default: 'p={mypid} u={user} |'
+  description: Logger name, as python formatted string.
+  env: [{name: ANSIBLE_LOG_NAME}]
+  ini:
+  - {key: log_name, section: defaults}
+  yaml: {key: defaults.log_format}
```
Comment (queria) [inline lib/ansible/config/base.yml:685]: maybe copy paste typo?  for LOG_NAME -> defaults.log_*format*?

File: lib/ansible/utils/display.py [hunk 2]
```diff
@@ -113,7 +113,7 @@ def display(self, msg, color=None, stderr=False, screen_only=False, log_only=Fal
         if color:
             msg = stringc(msg, color)
 
-        if not log_only:
+        if not log_only and color not in [C.COLOR_DEBUG]:
```
Comment (queria) [inline lib/ansible/utils/display.py:116]: seems like unrelated change?

**Review by queria — CHANGES_REQUESTED** [review]

File: lib/ansible/utils/display.py (hunk 2 above)
Comment (ssbarnea) [reply lib/ansible/utils/display.py:116]: True,  remaining from some testing.

File: lib/ansible/config/base.yml (hunk 1 above)
Comment (ssbarnea) [reply lib/ansible/config/base.yml:685]: True, copy/paste typo.

**Comment by ssbarnea** [comment]
@queria I am closing this because I used the wrong source branch by mistake. Please review https://github.com/ansible/ansible/pull/28663 which contains only this chance.
